### ISSUE TITLE
[Request] Support for multiple m3u's and xmltv sources?

### ISSUE THREAD

**Issue by CHJ85** [kodi-pvr/pvr.iptvsimple#358]
A few years ago, there was a fork of this pvr allowing you to add multiple sources.
It no longer works and is not being maintained, which is a shame.
Would it be possible to add this feature?
If I remember correctly, I believe the way this old fork worked was that it would cache the multiple playlists and xmltv files into just 1 m3u and 1 xml file each time you launched the TV guide.

**Comment by phunkyfish** [comment]
https://forum.kodi.tv/showthread.php?tid=340691

**Comment by CHJ85** [comment]
Idk if this is what I'm looking for.
This merges them into 1 at least, but that also means they won't be updated on-the-fly. Or does it merge whenever you open up the tv guide, or better yet more frequently while I'm in the guide?
Because I want the EPG to be up-to-date as well. And if it doesn't update, m3u8's with an expiration token simply won't work.
But I'll give it a try. Thanks.

**Comment by phunkyfish** [comment]
Yes, it does all that. I recommend using it on Kodi Matrix, there was a lot of work done to make it work well there. I can't speak to kodi 18, leia.

**Comment by CHJ85** [comment]
Thank you so much for your help.
It works beautify well.

**Comment by kekukui** [comment]
@CHJ85 - **Just for reference:** If you run multiple Kodi instances on the same LAN you may want to do this in TVheadend on a file server or NAS type device.  Then you can fetch the data just once and relay it to all of your clients, instead of having each Kodi instance grab duplicate live streams & EPG data.  This is especially useful if you run Kodi on a low end embedded Linux device like a Raspberry Pi.  Just install the TVheadend PVR client on Kodi once you have it configured.

**Comment by kvaster** [comment]
I've written iptv-proxy service for the purpose if someone is interested: https://github.com/kvaster/iptv-proxy

p.s.: also it allows to dynamically allocate/free iptv channel in case there are restrictions on iptv provider
